### DISCUSSION TITLE
1.2.0

### DISCUSSION THREAD

**Discussion by wlcrs** [wlcrs/huawei_solar#261] [Announcements]
# New features ✨ 

* Inverter ON/OFF switch
* [Capacity control mode](https://github.com/wlcrs/huawei_solar/wiki/Capacity-control)
* End-of-charge SOC and End-of-discharge SOC configuration entities
* Time of use periods and fixed charging/discharging periods can be retrieved through extended attributes of 'Time of use periods'  and 'Fixed charging periods' sensor
* Time of use periods are now editable through `huawei_solar.set_tou_periods` service
* Fixed charging periods are now editable through `huawei_solar.set_charging_periods` service
* Capacity control periods are editable through `huawei_solar.set_capacity_control_periods`

# Improvements
* Batched retrieval of configuration entity values
* Improved and dynamic correctness of minimum value/maximum value/availability of certain configuration entities (for example: capacity control can only be enabled if charging from grid is also enabled, capacity control SOC must be larger than maximum discharge SOC of the battery)
* Support for V103 Optimizer Data Files (fixes reading optimizer data in recent firmware versions) 
* Improve diagnostics
* Upgrade from pyModbus 2.5.3 to pyModbus 3.1.0

# Fixes
* Fix behavior when no config entities are available
* #198 Correctly process ValueError
* #199: Retry Modbus write commands on timeout or slave failure
* Fix computation of dynamic min/max value for number entities
* Fix bug preventing optimizers from loading
* #260: Increase timeout to 10 seconds
* #245: Fix 'native_unit_of_measurement' warnings

## Screenshots
![Inverter ON/OFF switch](https://user-images.githubusercontent.com/2150060/209295722-f8b3e303-67ee-4db9-ae0e-a1228517ccdb.png)
![Battery configuration](https://user-images.githubusercontent.com/2150060/209297503-171ea8c8-2945-452b-9762-c6fa3da45b47.png)
![TOU periods as extended attributes](https://user-images.githubusercontent.com/2150060/209295778-3edd33e9-cf3e-4fef-9359-0fe8ae6265d3.png)

**Full Changelog**: https://github.com/wlcrs/huawei_solar/compare/1.1.6...1.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/wlcrs/huawei_solar/releases/tag/1.2.0'>1.2.0</a>.</em>

**Comment by Vicente-Ch** [comment]
English 
Hello, I have a single-phase huawey sun2000-6ktl-L1 and last version Homeassistant.  I have updated to this version 1.2.0 (I come from the previous version) and the "inverter ON/OFF" switch does not appear.
 I was also interested in changing the work mode of the inverter, being able to go from maximum active power to export 0. If your integration does not have this option, could you tell me which Modbus registers I should write to do so?
 I would appreciate any clarification.
 Thanks. 

Spanish 
Hola, tengo un huawey sun2000-6ktl-L1 monofasico y Homeassistant ultima version. He actualizado a esta version 1.2.0 (vengo de la version justo anterior) y no me aparece el interruptor "inverter ON/OFF".
Ademas estaba interesado en  hacer cambiar de modo de trabajo al inversor, poder pasar de maxima potencia activa a exportacion 0. Si su integracion no dispone de esta opcion podria indicarme que registros Modbus debo escribir para hacerlo?. 
Agradeceria cualquier aclaracion. 
Gracias

**Comment by TMoersch** [comment]
Hi,
this is only available in Fusion Solar. If it is an installer-only option, I don't now.
I could set my export level on a sun2000-10ktl-L1 there to 0%.
Hope this is what you search. I did'nt see it in the integration, even in 1.2.1 its not integrated.
Thomas

**Reply by Vicente-Ch** [reply]
English:
The idea is to change the working mode of the inverter automatically depending on the needs of the home. This is necessary if you have a current rate that changes every hour; It changes the purchase price and also the sale price of surpluses.
I understand that it can be done. It seems that some people do it from nodered by writing in modbus, the issue is knowing what register I should write and what codes to set. This could also be possible with this great integration.

Español:
La idea es cambiar el modo de trabajo del inversor de forma automatica dependiendo de las necesidades del hogar. Esto es necesario si estas con una tarifa de corriente que cambia cada hora; cambia el precio de compra y tambien el precio de venta de excedentes.
Tengo entendido que se puede hacer, Parece ser que algunas personas lo hacen desde nodered escribiendo en modbus, el tema es saber que resgistro debo escribir y que codigos establecer.  Esto tambien podria ser posible con esta estupenda integracion.
